packages/langium/src/lsp/normalized-text-documents.ts

```diff
@@ -0,0 +1,217 @@
+/******************************************************************************
+ * Copyright 2024 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import type {
+    Connection, DidOpenTextDocumentParams, DidChangeTextDocumentParams, DidCloseTextDocumentParams, TextDocumentsConfiguration, TextDocumentChangeEvent,
+    TextDocumentWillSaveEvent, RequestHandler, TextEdit, Event, WillSaveTextDocumentParams, CancellationToken, DidSaveTextDocumentParams
+} from 'vscode-languageserver';
+import { TextDocumentSyncKind, Disposable, Emitter } from 'vscode-languageserver';
+import { UriUtils } from '../utils/uri-utils.js';
+
+export interface TextDocuments<T extends { uri: string }> {
```

Comment:
I think it makes sense to leave a hint on the origin of this definition, which is the class declaration in `vscode-languageserver/lib/common/textDocuments.d.ts`, right?